### ISSUE TITLE
Support links with href and meta (JSONAPI)

### ISSUE THREAD

**Issue by SeyZ** [emberjs/data#3588]
According to the spec http://jsonapi.org/format/#document-links

This kind of links should be handle by ember-data:

```
"links": {
  "related": {
    "href": "http://example.com/articles/1/comments",
    "meta": {
      "count": 10
    }
  }
}
```

At the moment, Ember-data complains with:

`You have pushed a record of type '...' with '...' as a link, but the value of that link is not a string.`

**Comment by wecc** [comment]
Thanks for reporting @SeyZ. We definitely want to support this.

**Comment by bartocc** [comment]
Has this issue been addressed by 2.0.0 by any chance ? 4f7eedd19b5e9481ff485069cf11c26739091db9
